.gitignore

```diff
@@ -5,53 +5,73 @@
 # created during manual debugging and many people like to clean up instead of
 # having git ignore such leftovers. You can use `.git/info/exclude` to
 # configure your local ignore list.
-# FIXME: This needs cleanup.
-*~
+
+# File system
+.DS_Store
+desktop.ini
+
+# Version control
+.hgignore
+.hg/
+
+# Editor
 *.swp
 *.swo
-.#*
-.DS_Store
+Session.vim
 .cproject
-.hg/
-.hgignore
 .idea
 *.iml
-__pycache__/
-*.py[cod]
-*$py.class
+.vscode
 .project
+.favorites.json
 .settings/
+
+# Tool
 .valgrindrc
-.vscode
-.favorites.json
-/Makefile
-/build/
+.cargo
+# Included because it is part of the test case
+!/src/test/run-make/thumb-none-qemu/example/.cargo
+
+# Configuration
 /config.toml
-/dist/
+/Makefile
+config.mk
+config.stamp
+no_llvm_build
+
+# Build
 /dl/
 /doc/
 /inst/
 /llvm/
 /mingw-build/
-/src/tools/x/target
-# Created by default with `src/ci/docker/run.sh`:
-/obj/
-/unicode-downloads
-/target
-# Generated by compiletest for incremental:
 /tmp/
+/build/
+/dist/
+/obj/
+/unicode-downloads/
+/target/
```

Comment:
I added a trailing slash for consistency, or does that change anything?

Reply:
>           If there is a separator at the end of the pattern then the pattern will
           only match directories, otherwise the pattern can match both files and
           directories.

I think only matching directories is correct.

Reply:
Dropping a trailing slash is needed to support symlinks, and that's what cargo generates by default.
I recommend reverting the addition of a trailing slash.

See also https://github.com/rust-lang/cargo/pull/4944, https://github.com/rust-lang/rust/pull/70409

> When target is a symlink (e.g., to keep build outputs on a separate
> partition), Git will not match the /target/ ignore to the symlink
> since it is not a directory. Drop the trailing slash to support ignoring
> target as a symlink.